### PR TITLE
Replace zxcvbn by zxcvbn-ts

### LINKED ISSUE #16974: Switch to zxcvbn-ts
Because of https://github.com/dropbox/zxcvbn/issues/290 
The zxcvbn project did not receive updates since 2017.

We should now use: https://github.com/zxcvbn-ts/zxcvbn

### PR DESCRIPTION
**Pull request by Guileas**
Signed-off-by: Gwénola ETHEVE <git@etheve.eu>

### Description

Replace zxcvbn by zxcvbn-ts

Fixes #16974

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,9 +14,7 @@
     "@babel/core": "^7.14.6",
     "@babel/preset-env": "^7.14.7",
     "@popperjs/core": "^2.9.2",
-    "autoprefixer": "^10.2.6",
-    "bootstrap": "5.0.2",
-    "codemirror": "5.62.0",
```
Comment (mauriciofauth) [inline package.json:19]: I think removing these dependencies was a mistake.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Guileas) [reply package.json:19]: My bad ^^

**Review by williamdes — APPROVED** [review]
Very good work 💯

File: package.json [hunk 2]
```diff
@@ -38,7 +39,6 @@
     "tracekit": "0.4.6",
     "u2f-api-polyfill": "0.4.4",
     "updated-jqplot": "1.0.9",
```
Comment (mauriciofauth) [inline package.json:41]:
```suggestion
    "updated-jqplot": "1.0.9"
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by williamdes** [comment]
@sudo-robot deploy

**Comment by MrWook** [comment]
Hey, i really recommend that you are using the recommended set of options otherwise the scoring is pretty bad.

```
import { zxcvbn, ZxcvbnOptions } from '@zxcvbn-ts/core'
import zxcvbnCommonPackage from '@zxcvbn-ts/language-common'
import zxcvbnEnPackage from '@zxcvbn-ts/language-en'

const options = {
  graphs: zxcvbnCommonPackage.adjacencyGraphs,
  dictionary: {
    ...zxcvbnCommonPackage.dictionary,
    ...zxcvbnEnPackage.dictionary,
  },
}

ZxcvbnOptions.setOptions(options)
```

**Comment by williamdes** [comment]
To achieve this you will need the sam technique as #16959 with webpack, to re-bundle the js bundle

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
> To achieve this you will need the sam technique as #16959 with webpack, to re-bundle the js bundle

This can be done in a different pull request.

**Pull request merged**